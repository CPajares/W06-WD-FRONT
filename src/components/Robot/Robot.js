import useRobot from "../../hooks/useRobot";

const Robot = ({ robot, day, month, year }) => {
  const { deleteRobotById } = useRobot();

  const onClickDelete = () => {
    deleteRobotById(robot._id);
  };

  return (
    <>
      <section className="container-robot col-6">
        <div className="col-12">
          <h2 className="bg-info">Nombre: {robot.name}</h2>
          <img
            src={robot.url}
            width={200}
            height={200}
            alt={`${robot.name}`}
          ></img>
          <h3>Caracteristicas</h3>
          <ul className="list-group row">
            <li className="list-group-item list-group-item-success col-10">
              Velocidad:{robot.features.speed}
            </li>
            <li className="list-group-item list-group-item-success col-10">
              Resistencia: {robot.features.resistence}
            </li>
            <li className="list-group-item list-group-item-success col-10">
              Fecha: {day}/{month}/{year}
            </li>
          </ul>
          <button type="button" className="btn btn-warning col-3">
            MODIFICAR
          </button>
          <button className="btn btn-danger col-3" onClick={onClickDelete}>
            BORRAR
          </button>
        </div>
      </section>
    </>
  );
};

export default Robot;
