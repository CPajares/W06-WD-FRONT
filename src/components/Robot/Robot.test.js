import Robot from "./Robot";
import ReactTestRenderer from "react-test-renderer";

describe("Give a Robot component", () => {
  describe("When it receive an object robot", () => {
    test("Then should render a Card with the name", () => {
      const robotObject = {
        _id: "618678658516adffbffcac4b",
        name: "Bender",
        url: "https:",
        features: {
          speed: 2,
          resistence: 3,
          date: "2021-02-04T23:00:00.000Z",
        },
      };
      const robotCard = ReactTestRenderer.create(<Robot robot={robotObject} />);

      expect(robotCard.toJSON()).toMatchSnapshot();
    });
  });
});
