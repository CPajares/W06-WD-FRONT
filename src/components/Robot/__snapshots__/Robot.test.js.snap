// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Give a Robot component When it receive an object robot Then should render a Card with the name 1`] = `
Array [
  <h2>
    Nombre: 
    Bender
  </h2>,
  <img
    alt="Bender"
    src="https:"
  />,
  <h3>
    Caracteristicas
  </h3>,
  <ul
    className="list-group"
  >
    <li
      className="list-group-item list-group-item-success"
    >
      Velocidad:
      2
    </li>
    <li
      className="list-group-item list-group-item-success"
    >
      Resistencia: 
      3
    </li>
    <li
      className="list-group-item list-group-item-success"
    >
      Fecha: 
      /
      /
    </li>
  </ul>,
  <button
    className="btn btn-warning"
    type="button"
  >
    MODIFICAR
  </button>,
  <button
    className="btn btn-danger"
  >
    BORRAR
  </button>,
]
`;
